src/system.py

```diff
@@ -37,134 +37,135 @@ class System:
     def __init__(self, repo_path=None, definitions=None, programs=None):
         '''
         A system (collection of programs) can be created in at least 4 ways
-        
+
         - (repo_path) Local directory containing git checkouts of programs
         - (definitions) Baserock stratam file
         - (programs) List of Program instances
         - Empty system - Programs to add later
-        
+
         @param repo_path String Path to the directory that contains the system
         @param definitions String Path to definitions file that describes the system
         @param program Program[] List of programs in a system
         '''
-        
+
         # Check that more than one entry hasn't been given
         if [repo_path, definitions, programs].count(None) < 2:
             logger.error(list2str([repo_path, definitions, programs]))
             raise TypeError(u"System expects ONLY a repo_path, definitions or programs, parameter")
-        
+
         # Scan directory for programs (git)
         if repo_path is not None:
             logger.debug("Creating System from repo_path")
             self.programs=System.programsFromGitDirectory(repo_path)
-            
+
         # Parse definitions
         elif definitions is not None:
             #TODO
             logger.debug("Creating System from baserock definitions")
             self.programs=[]
-            
+
         # List of program instances given
         elif programs is not None:
             #TODO check that this is a list and all members are Programs
             logger.debug("Creating System from list")
             self.programs=programs
-        
+
         # Empty system, expected to use addProgram() after creation
         else:
             logger.debug("Creating an empty System")
             self.programs=[]
-    
-        logger.debug("Program list for this system: %s"%list2str(self.programs))
```

Comment:
Unless python3 has corrected this, printing a list (tostring) is not recursive and will just return [<Program>, <Program>, ...]

I will need to double check this